### ISSUE TITLE
Make GitHub properly detect and highlight Coconut source

### ISSUE THREAD

**Issue by evhub** [evhub/coconut#335]
See <https://github.com/github/linguist>.

**Comment by evhub** [comment]
In progress at https://github.com/github/linguist/pull/3872.

**Comment by eindiran** [comment]
I was reading over the [guidelines](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language) for accepting a language into linguist and it appears that there must be 'hundreds' of unique repositories in the wild.

If you look for [Coconut usage in the wild](https://github.com/search?utf8=%E2%9C%93&q=extension%3Acoco+NOT+nothack&type=Code), I don't think it is currently sufficient. There are dozens of unique repositories, but not yet hundreds.

In the meantime, would it be worthwhile to come up with a grammar for Coconut that isn't GPL'ed? The linguist repo doesn't accept grammars which are, but I saw that the linked grammar in the [PR](https://github.com/github/linguist/pull/3872) you opened @evhub is GPL'ed.

**Comment by evhub** [comment]
@eindiran Yeah, developing a non-GPL grammar would definitely be a good idea. A possible starting point would be to adapt [MagicPython](https://github.com/MagicStack/MagicPython).

**Comment by Darkle** [comment]
As a fallback, you can use the .gitattributes override feature in individual repos: https://github.com/github/linguist#using-gitattributes
